controllers/activemqartemis_jdbc_ha_test.go

```diff
@@ -0,0 +1,283 @@
+
+				By("deploying artemis")
+
+				peerLabel := "fast-ha-peer"
+				peerPrefix := NextSpecResourceName()
+				ctx := context.Background()
+				peerA := generateArtemisSpec(defaultNamespace)
+				peerA.Name = peerPrefix + "-peer-a"
+
+				peerA.Spec.Acceptors = []brokerv1beta1.AcceptorType{{Name: "tcp", Port: 61616, Expose: true}}
+
+				if !isOpenshift {
+					peerA.Spec.IngressDomain = defaultTestIngressDomain
+				}
+
+				peerA.Spec.Console.Expose = true
+
+				peerA.Spec.DeploymentPlan.PersistenceEnabled = boolFalse
+				peerA.Spec.DeploymentPlan.Clustered = &boolFalse
+
+				peerA.Spec.DeploymentPlan.Labels = map[string]string{peerLabel: peerPrefix}
+
+				peerA.Spec.Env = []corev1.EnvVar{
+					{
+						Name:  "ARTEMIS_EXTRA_LIBS",
+						Value: "/amq/init/config/extra-libs",
+					},
+				}
+
+				By("configuring the broker")
+				peerA.Spec.BrokerProperties = []string{
+					"addressConfigurations.JOBS.routingTypes=ANYCAST",
+					"addressConfigurations.JOBS.queueConfigs.JOBS.routingType=ANYCAST",
+
+					"HAPolicyConfiguration=SHARED_STORE_PRIMARY",
+					"storeConfiguration=DATABASE",
+					"storeConfiguration.jdbcDriverClassName=org.postgresql.Driver",
+					"storeConfiguration.jdbcConnectionUrl=jdbc:postgresql://" + dbName + ":" + fmt.Sprintf("%d", dbPort) + "/postgres?user=postgres&password=postgres",
+				}
+
+				By("patching ss to add init container to pull down jdbc jar")
+
+				kindMatchSs := "StatefulSet"
+				peerA.Spec.ResourceTemplates = []brokerv1beta1.ResourceTemplate{
+					{
+						Selector: &brokerv1beta1.ResourceSelector{
+							Kind: &kindMatchSs,
+						},
+						Patch: &unstructured.Unstructured{Object: map[string]interface{}{
+							"kind": "StatefulSet",
+							"spec": map[string]interface{}{
+								"template": map[string]interface{}{
+									"spec": map[string]interface{}{
+										"initContainers": []interface{}{
+											map[string]interface{}{
+												"name":  "download-jdbc-driver",
+												"image": version.LatestInitImage,
+												"command": []interface{}{
+													"sh",
+													"-c",
+													"mkdir -p /amq/init/config/extra-libs && curl -o /amq/init/config/extra-libs/postgresql-42.7.1.jar https://jdbc.postgresql.org/download/postgresql-42.7.1.jar",
+												},
+												"volumeMounts": []interface{}{
+													map[string]interface{}{
+														"name":      "amq-cfg-dir",
+														"mountPath": "/amq/init/config",
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						},
+					},
+				}
+
+				peerB := &brokerv1beta1.ActiveMQArtemis{}
+
+				// a clone of peerA
+				peerA.DeepCopyInto(peerB)
+				peerB.Name = peerPrefix + "-peer-b"
+
+				By("provisioning the broker peer-a")
+				Expect(k8sClient.Create(ctx, &peerA)).Should(Succeed())
+
+				By("provisioning the broker peer-b")
+				Expect(k8sClient.Create(ctx, peerB)).Should(Succeed())
```

Comment:
One of the peers crashes because of the readiness probe:
```
Readiness probe failed:
Back-off restarting failed container aa-jdbc-ha51a-peer-a-container in pod aa-jdbc-ha51a-peer-a-ss-0_test(2630511f-9729-4b6a-b477-0b636ab5a194)
```

Reply:
exactly, we need to avoid that. we can only stay in the starting phase for max 300s so a startup probe wont cut it.
I think we need a liveness probe that knows about the waiting to lock sate and is happy to report live in that state and a readiness probe that gates the network access in the normal way.

I think we can build these over the broker logging of specific events, and can look at broker status via a jolokia agent at a later date.